plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1161,44 +1165,76 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         BaseTable icebergTable = (BaseTable) catalog.loadTable(session, table.getSchemaTableName());
 
+        Set<String> unsupportedProperties = Sets.difference(properties.keySet(), UPDATABLE_PROPERTIES);
+        if (!unsupportedProperties.isEmpty()) {
+            throw new TrinoException(NOT_SUPPORTED, "The following properties cannot be updated: " + String.join(", ", unsupportedProperties));
+        }
+
         transaction = icebergTable.newTransaction();
         UpdateProperties updateProperties = transaction.updateProperties();
 
-        for (Map.Entry<String, Optional<Object>> propertyEntry : properties.entrySet()) {
-            String trinoPropertyName = propertyEntry.getKey();
-            Optional<Object> propertyValue = propertyEntry.getValue();
+        Optional<Object> fileFormat = properties.get(FILE_FORMAT_PROPERTY);
+        if (fileFormat != null && fileFormat.isPresent()) {
```

Comment:
i understand per https://github.com/trinodb/trino/pull/12259#discussion_r874936281 that `fileFormat.isPresent()` is redundant
```java
if (fileFormat != null) {
 verify(fileFormat.isPresent(), "fileFormat cannot be empty");
```